tests/Fixtures/TestBundle/Document/DummyDifferentSerializationGroup.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Tests\Fixtures\TestBundle\Document;
+
+use ApiPlatform\Core\Annotation\ApiResource;
+use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;
+use Symfony\Component\Serializer\Annotation\Groups;
+
+/**
+ * Dummy With different serialization groups for item_query and collection_query.
+ *
+ * @author Mahmood Bazdar <mahmood@bazdar.me>
+ *
+ * @ApiResource(
+ *     graphql={
+ *         "item_query"={"normalization_context"={"groups"={"item_query"}}},
+ *         "collection_query"={"normalization_context"={"groups"={"collection_query"}}}
+ *     }
+ * )
+ * @ODM\Document
+ */
+class DummyDifferentSerializationGroup
```

Comment:
```suggestion
class DummyDifferentGraphQlSerializationGroups
```

Reply:
I changed the class name to  `DummyDifferentGraphQlSerializationGroup` because if the class name is already pluralized and array key will be the same for item and collections because we don't singularize it for items:

https://github.com/api-platform/core/blob/78478a75b3b02c6fef4224b563913a076ad5e246/src/GraphQl/Type/FieldsBuilder.php#L102

https://github.com/api-platform/core/blob/78478a75b3b02c6fef4224b563913a076ad5e246/src/GraphQl/Type/FieldsBuilder.php#L122


I can change it now but I think this should be another PR.

Reply:
It's not done in REST, so I'm not sure if we should do it for GraphQL.

Reply:
But this cause error(500) and I think we should fix it.

Reply:
OK then.